packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -622,22 +622,34 @@ export class Cluster extends Resource implements ICluster {
       description: 'EKS Control Plane Security Group',
     });
 
+    const controlPlanePort = ec2.Port.tcp(443); // Control Plane has an HTTPS API
+
     this.connections = new ec2.Connections({
       securityGroups: [securityGroup],
-      defaultPort: ec2.Port.tcp(443), // Control Plane has an HTTPS API
+      defaultPort: controlPlanePort,
     });
 
     this.vpcSubnets = props.vpcSubnets ?? [{ subnetType: ec2.SubnetType.PUBLIC }, { subnetType: ec2.SubnetType.PRIVATE }];
 
     // Get subnetIds for all selected subnets
     const subnetIds = [...new Set(Array().concat(...this.vpcSubnets.map(s => this.vpc.selectSubnets(s).subnetIds)))];
 
+    this.kubctlProviderSecurityGroup = new ec2.SecurityGroup(this, 'KubectlProviderSecurityGroup', {
+      vpc: this.vpc,
+      description: 'Comminication between KubectlProvider and EKS Control Plane',
+    });
+
+    this.kubctlProviderSecurityGroup.addIngressRule(this.kubctlProviderSecurityGroup, controlPlanePort);
+
     const clusterProps: CfnClusterProps = {
       name: this.physicalName,
       roleArn: this.role.roleArn,
       version: props.version.version,
       resourcesVpcConfig: {
-        securityGroupIds: [securityGroup.securityGroupId],
+        securityGroupIds: [
+          securityGroup.securityGroupId,
+          this.kubctlProviderSecurityGroup.securityGroupId,
```

Comment:
as discussed, this is a breaking change and should be merged into our dev-preview staging branch